init.c

```diff
@@ -1776,36 +1776,31 @@ static int parse_unalias(struct Buffer *buf, struct Buffer *s,
     {
       if (CurrentMenu == MENU_ALIAS)
       {
-        for (tmp = Aliases; tmp; tmp = tmp->next)
-          tmp->del = true;
+        TAILQ_FOREACH(a, &Aliases, entries)
+        {
+          a->del = true;
+        }
         mutt_menu_set_current_redraw_full();
       }
       else
-        mutt_alias_free(&Aliases);
+        mutt_aliaslist_free(&Aliases);
       break;
     }
     else
     {
-      for (tmp = Aliases; tmp; tmp = tmp->next)
+      TAILQ_FOREACH_SAFE(a, &Aliases, entries, tmp)
       {
-        if (mutt_str_strcasecmp(buf->data, tmp->name) == 0)
+        if (mutt_str_strcasecmp(buf->data, a->name) == 0)
         {
           if (CurrentMenu == MENU_ALIAS)
           {
-            tmp->del = true;
+            a->del = true;
             mutt_menu_set_current_redraw_full();
-            break;
           }
-
-          if (last)
-            last->next = tmp->next;
           else
-            Aliases = tmp->next;
-          tmp->next = NULL;
-          mutt_alias_free(&tmp);
+            mutt_alias_free(a);
```

Comment:
I think we need a `REMOVE` here

Reply:
Fixed. Also eliminated `tmp` variable, because we don't need `_SAFE` macro here (looping stops after element removal).